typing/btype.ml

```diff
@@ -628,22 +628,40 @@ let is_optional_parsetree : Parsetree.arg_label -> bool = function
 type optional_module_path = Stdlib_option | Stdlib_or_null
 
 let classify_module_path : Longident.t -> optional_module_path = function
-  | Ldot(Lident "Stdlib", "Option") -> Stdlib_option
-  | Ldot(Lident "Stdlib", "Or_null") -> Stdlib_or_null
+  | Lident "option" -> Stdlib_option
+  | Lident "or_null" -> Stdlib_or_null
   | _ -> failwith "Only expected Stdlib.Option and Stdlib.Or_null"
 
 type optionality = Optional_arg of optional_module_path
                  | Required_or_position_arg
 
 let classify_optionality (lbl: Types.arg_label) = match lbl with
   | Optional _ -> Optional_arg Stdlib_option
-  | Generic_optional(path, _) -> Optional_arg (classify_module_path path.txt)
+  | Generic_optional(path, _) -> Optional_arg (classify_module_path path)
   | Labelled _ | Position _ | Nolabel -> Required_or_position_arg
 
-let classify_optionality_parsetree (lbl : Parsetree.arg_label) = match lbl with
+let extract_optional_tp_from_pattern_constraint pat = match pat with
+  | {Parsetree.ppat_desc = Ppat_constraint (under_pat, ty, modes); _} ->
+    (match ty with
+    | Some {ptyp_desc = Ptyp_constr ({ txt = Lident ident_name}, [arg]); _} ->
```

Comment:
This pattern-match can be merged with the previous by expanding `ty` correspondendgly.